### PR TITLE
fix: support for CDATA in oembed

### PR DESCRIPTION
**Pull request by jacktuck**

### CODE REVIEW

**Comment by jacktuck** [comment]
Test

```
❯ curl "https://unfurljs.herokuapp.com/?url=https://soundcloud.com/the-bugle" -s | jq '.oEmbed'
{
  "version": "1.0",
  "type": "rich",
  "height": 450,
  "width": 100,
  "title": "The Bugle",
  "html": "<iframe width=\"100%\" height=\"450\" scrolling=\"no\" frameborder=\"no\" src=\"https://w.soundcloud.com/player/?visual=true&url=https%3A%2F%2Fapi.soundcloud.com%2Fusers%2F9818871&show_artwork=true\"></iframe>"
}

```

**Comment by jacktuck** [comment]
@adrian-seijo let me know what you think.

File: src/index.ts [hunk 1]
```diff
@@ -147,71 +149,91 @@ function getRemoteMetadata(ctx) {
...
+              parserContext.tagName = "";
+              parserContext.text = "";
+            },
+            onend: function () {
+              resolve(content);
+            },
           },
-        });
+          {
+            recognizeCDATA: true,
```
Comment (jacktuck) [inline src/index.ts:234]: Tells 'htmlparser` to look for ctags

File: src/index.ts [hunk 2]
```diff
@@ -147,71 +149,91 @@ function getRemoteMetadata(ctx) {
...
+
       ret = await new Promise((resolve) => {
-        const parser = new Parser({
-          onopentag: function (name, attribs) {
-            if (this._is_html) {
-              if (!content.html) {
-                content.html = "";
+        const parser = new Parser(
+          {
+            oncdataend: () => {
```
Comment (jacktuck) [inline src/index.ts:171]: Now we pass `recognizeCDATA` we can add a `oncdataend` handler which will get called when the ctag closes.

**Comment by adrian-seijo** [comment]
> src/index.ts

Definetively a better approach, love that 5 major version bump of the parser 😜 Thanks for the release!

**Pull request merged**